odpc.client/src/features/waardelijst/service.ts

```diff
@@ -1,28 +1,41 @@
-import { computed } from "vue";
-import { useFetchApi } from "@/api/use-fetch-api";
-import {
-  WAARDELIJSTEN,
-  type GroupedWaardelijstItems,
-  type Waardelijst,
-  type WaardelijstItem
-} from "./types";
-
-const {
-  data,
-  isFetching: loadingWaardelijstItems,
-  error: waardelijstItemsError
-} = useFetchApi(`/api/v1/waardelijsten`).json<WaardelijstItem[]>();
-
-const waardelijstIds = computed(() => data.value?.map((item) => item.id) || []);
-
-const groupedWaardelijstItems = computed<GroupedWaardelijstItems>(() =>
-  Object.keys(WAARDELIJSTEN).reduce((result: GroupedWaardelijstItems, key) => {
-    result[key as Waardelijst] =
-      data.value
-        ?.filter((item) => item.type === key)
-        ?.sort((a, b) => a.name.localeCompare(b.name)) || [];
-    return result;
-  }, {} as GroupedWaardelijstItems)
+import { computed, ref } from "vue";
+import { useFetchApi, type PagedResult } from "@/api";
+import type { WaardelijstItem } from "./types";
+
+const API_URL = `/api/v1`;
+
+const organisaties = ref<WaardelijstItem[]>([]);
+const informatiecategorieen = ref<WaardelijstItem[]>([]);
+
+const loadingWaardelijsten = ref(false);
+const waardelijstenError = ref<string>();
+
+(async () => {
```

Comment:
gaat zichzelf laden? is dat intuitief?

Reply:
Dit is een [immediately invoked function expression](https://developer.mozilla.org/en-US/docs/Glossary/IIFE#execute_an_async_function). Die kan je gebruiken om async code direct uit te voeren in browsers die geen 'top level await' ondersteunen. Aangezien de [browser support](https://caniuse.com/?search=top%20level%20await) daarvoor echter prima is, kunnen we wat mij betreft top level await gebruiken.

Reply:
ik snap hoe het werkt, maar vanuit de consumerende componenten in de rest van de applicatie gezien is dit niet heel transparant en het wijkt af van de patterns die elders gebruikt worden

Reply:
8d322ea354fe8016e9d1ab6f955c5718a6b67788